[ANDROID] List Dialogs, Guest Options Refactor and Empty List Pages Design

asliayk opened the pull request:
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other

# Description

- For list selection, I created two dialogs. After pressing the save button on `ProductFragment`, a dialog opens with her lists as radio buttons, and there is also a "Create New List" button on it, which opens another new dialog with an edittext on it. So, the user can type a list name to create a new list.
- For a more realistic app design, instead of showing empty pages and only a floating action button when the user has no credit cards, addresses or product lists, I created three empty page designs for `ListsFragment`, `AddressFragment` and `BankAccountFragment` with an image view, text view and add button.
- I disabled navigating to the complete order page if the shopping cart is empty. Instead of navigating, a dialog opens which warns the user as "_You don't have any products in your shopping cart._".
- There were still some mock data on the code, so I removed them.
- I removed the green borders on floating action buttons.
- I also added the navigation from `ProductFragment` to `VendorPageFragment`.

When the user does not log in and enters the app as a guest, there were some options that shouldn't be available in order not to make the app crashed:
- **Navigating to the complete order page:** To make it disabled for guest, I added a guest check, which opens a dialog that warns the user as "_You should be logged in to make a purchase_.".
- **Adding products to shopping cart:** To make it disabled for guest, I added a guest check, which opens a dialog that warns the user as "_You should be logged in to add product to your shopping cart._".
- **Navigating to the bank accounts page**: To make it disabled for guest, I added a guest check, which opens a dialog that warns the user as "_You should be logged in to see your bank account info._".

# Screenshots

#### List Selection & Add New List Dialogs
<p float="left">
<img width="181" alt="list dialog" src="https://user-images.githubusercontent.com/56061820/105247157-6056a080-5b85-11eb-9100-cd986eeddd51.PNG">
<img width="179" alt="addlistdialog" src="https://user-images.githubusercontent.com/56061820/105247173-664c8180-5b85-11eb-9f23-0be428408cce.PNG">
</p>

#### Empty Versions of AddressFragment, BankAccountFragment and ListsFragment
<p float="left">
<img width="181" alt="emptyaddress" src="https://user-images.githubusercontent.com/56061820/105247295-9dbb2e00-5b85-11eb-9716-fcf6bd875c97.PNG">
<img width="180" alt="emptycreditcard" src="https://user-images.githubusercontent.com/56061820/105247305-a27fe200-5b85-11eb-9c3a-23bd1841aaf3.PNG">
<img width="181" alt="emptylist" src="https://user-images.githubusercontent.com/56061820/105247318-a6136900-5b85-11eb-9879-86de81a7e285.PNG">
</p>


# Test results

No tests for now.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have checked my code and corrected any misspellings
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Potential problems/Important remarks

asliaykold committed:
purchase options disabled for guest

asliaykold committed:
accept order with empty shopping cart bug resolved

asliaykold committed:
visibilities edited, fab borders removed

asliaykold committed:
list selection alert dialog added

asliaykold committed:
guest button height increased

asliaykold committed:
list selection dialogs added

asliaykold committed:
empty lists page design added

asliaykold committed:
fab button to listsfragment added

asliaykold committed:
click listener for fab button added

asliaykold committed:
empty back accounts design added

asliaykold committed:
empty address layout added

asliaykold committed:
vendor page navigation added

karalleyna reviewed (commented):
Have you noticed that the float action button is shown on the bottom right of the fragment in small time frame when you navigate from the profile page of the customer to his or her bank account page that is currently empty? Is it what you wanted or a possible issue that should be handled?

https://user-images.githubusercontent.com/36455180/105521526-18eb2400-5ced-11eb-8e5c-0d72c121c5bd.mov

Other than this issue, the difference between the colors of the icons in fragments for empty lists is just 
a fly in the ointment. So, I would like to be happy if you remove this distinction. Overall, it seems correct.

asliaykold committed:
loading states added & visibility changes updated

asliaykold committed:
gold colored icon created

asliayk commented:
> Have you notice that the float action button is shown on the bottom right of the fragment in small time frame when you navigate from the profile page of the customer to his or her bank account page that is currently empty? Is it what you wanted or a possible issue that should be handled?
> 
>  pr1_change.mov 
> Other than this issue, the difference between the colors of the icons in fragments for empty lists is just
> a fly in the ointment. So, I would like to be happy if you remove this distinction. Overall, it seems correct.

- I wasn't aware of this visibility issue. In order to solve it, I added loading animations to credit carts and addresses pages, and updated the visibility of fab button as _GONE_ in an earlier stage, so no fab buttons appear before loading data from API.
- I also added a gold colored icon to the lists page.
